test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+          errorArgs: _*
+        )
+        doc.title() should startWith("Error:")
+      },
+      BAD_REQUEST
+    )
+  }
+
+  "Supporting Evidence Controller" when {
+
+    "handling requests to display the do you want to upload supporting evidence page" must {
+
+      def performAction(): Future[Result] = controller.doYouWantToUploadSupportingDocuments()(FakeRequest())
+
+      "display the page" when {
+
+        "a user has not answered the question before" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithSingleDisposalState(
+                None
+              )._1
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey("supporting-evidence.do-you-want-to-upload.title"), { doc =>
+              doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.SupportingEvidenceController
+                .doYouWantToUploadSupportingDocuments()
+                .url
+            }
+          )
+        }
+
+        "a user has answered the question before" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithSingleDisposalState(
+                Some(IncompleteSupportingEvidenceAnswers(Some(false), List.empty, List.empty))
+              )._1
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey("supporting-evidence.do-you-want-to-upload.title"), { doc =>
+              doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.SupportingEvidenceController
+                .doYouWantToUploadSupportingDocuments()
+                .url
+            }
+          )
+        }
+      }
+    }
+
+    "handling submission of do you want to upload supporting evidence page" must {
+
+      def performAction(data: (String, String)*): Future[Result] =
+        controller.doYouWantToUploadSupportingDocumentsSubmit()(FakeRequest().withFormUrlEncodedBody(data: _*))
+
+      "display the technical error page" when {
+
+        "an error occurs when storing the draft return and answer is true" in {
+          val answers =
+            IncompleteSupportingEvidenceAnswers.empty.copy(
+              doYouWantToUploadSupportingEvidence = Some(true),
+              List.empty,
+              List.empty
+            )
+
+          val (session, journey, draftReturn) = sessionWithSingleDisposalState(Some(answers))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              draftReturn,
```

Comment:
we don't need to bother making a call to the BE to store a draft return if nothing has changed